### ISSUE TITLE
Dialog should have an option to put the close button last in the pdomOrder

### ISSUE THREAD

**Issue by zepumph** [phetsims/sun#724]
While discussing https://github.com/phetsims/joist/issues/750 in design meeting, we thought that some dialogs would seem better to have the close button last. It seems like a good use case to support, and we likely will for in the vegas reward dialog (over in https://github.com/phetsims/vegas/issues/96). 

Tagging @jessegreenberg because he is a boss, but I'll take the lead first.

**Comment by zepumph** [comment]
@jessegreenberg and I like `closeButtonLastInPDOMOrder` defaulting to false.

I also noticed that the current PDOM order can leak a `null` into it if the title is not provided. Let's make sure we get rid of that.

**Comment by zepumph** [comment]
@jessegreenberg please review this. I added assertions because I had to do some hackary to come up with a good default focusOnShowNode when the close button is last.

**Comment by zepumph** [comment]
Hows that?


Tested with:

```diff
Index: js/AboutDialog.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/AboutDialog.js b/js/AboutDialog.js
--- a/js/AboutDialog.js	(revision cd9dc495092421afbbf827ff0c9dbf5888fa09ec)
+++ b/js/AboutDialog.js	(date 1634251723326)
@@ -45,6 +45,7 @@
       xSpacing: 26,
       topMargin: 26,
       bottomMargin: 26,
+      closeButtonLastInPDOM: true,
       leftMargin: 26,
       rightMargin: 26,
       phetioReadOnly: true, // the AboutDialog should not be settable

```

**Comment by jessegreenberg** [comment]
I think thats great, thanks!
